From 9dd546bdd701ac293525851b695a180bdb54524c Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Mon, 2 Jul 2018 12:51:51 +0200
Subject: [PATCH 0664/1051] arm64: dts: marvell: ap810: reserve 16M memory for
 tee

Reserve 16M memory for TEE, starts from 0x4400000.

These memory are used for TEE code, data, communication channel
and shared memory with Non-Trusted world.

For the TEE code and data section, it will be configured inaccessible
to Non-Trusted World by memory controller.

The rest of memory are either used for communication or shared memory
between Trusted-World and Non-trusted World, which is TEE implementation
defined and may be managed by the TEE driver

Patch based on Zhoujie Wu <zjwu@marvell.com> commit:
"dts: armada-ap806: reserve 16M memory for tee"

Change-Id: I8d5a407fce1e5bf86dea9474864c4f3c26c94b9e
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap810-ap0.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap810-ap0.dtsi b/arch/arm64/boot/dts/marvell/armada-ap810-ap0.dtsi
index 00618cc1bf2d..ca9e4c8013da 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap810-ap0.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap810-ap0.dtsi
@@ -30,6 +30,11 @@
 			no-map;
 		};
 
+		tee@4400000 {
+			reg = <0 0x4400000 0 0x1000000>;
+			no-map;
+		};
+
 		maintenance-area@8000000 {
 			reg = <0 0x8000000 0 0x100000>;
 			no-map;
-- 
2.17.1

